### PR TITLE
tidb-server: Add domain close and adjust `exit` position

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: tidb-server/main.go [hunk 1]
```diff
@@ -161,7 +161,6 @@ func main() {
 		sig := <-sc
 		log.Infof("Got signal [%d] to exit.", sig)
 		svr.Close()
-		os.Exit(0)
```
Comment (shenli) [inline tidb-server/main.go]: Should we call domain.Close() here?

Comment (zimulala) [inline tidb-server/main.go]: If put it here, we can't make sure it be invoked at the end(this logic is in goroutine).

File: tidb-server/main.go [hunk 2]
```diff
@@ -172,6 +171,8 @@ func main() {
 	pushMetric(*metricsAddr, time.Duration(*metricsInterval)*time.Second)
 
 	log.Error(svr.Run())
+	domain.Close()
```
Comment (tiancaiamao) [inline tidb-server/main.go]: Are you sure TiDB can run to this line? @zimulala

Comment (zimulala) [inline tidb-server/main.go]: Yep

**Comment by tiancaiamao** [comment]
LGTM

File: tidb-server/main.go (hunk 2 above)
Comment (shenli) [inline tidb-server/main.go]: I have tested stop tidb-server with `ctrl-c` and `kill -1`. I am sure TiDB will run to this line.

**Pull request merged**